tests/expressions/tests.py

```diff
@@ -205,6 +206,100 @@ def test_update(self):
             ],
         )
 
+    def _test_slicing_of_f_expressions(self, model):
+        tests = [
+            (F("name")[:], "Example Inc.", "Example Inc."),
+            (F("name")[:7], "Example Inc.", "Example"),
+            (F("name")[:6][:5], "Example", "Examp"),  # Nested slicing.
+            (F("name")[0], "Examp", "E"),
+            (F("name")[5], "E", ""),
+            (F("name")[7:], "Foobar Ltd.", "Ltd."),
+            (F("name")[0:10], "Ltd.", "Ltd."),
+            (F("name")[2:7], "Test GmbH", "st Gm"),
+            (F("name")[1:][:3], "st Gm", "t G"),
+            (F("name")[2:2], "t G", ""),
+        ]
+        for expression, name, expected in tests:
+            with self.subTest(expression=expression, name=name, expected=expected):
```

Comment:
This doesn't look like an appropriate use of `subTest`. Isn't each iteration dependent upon the previous iteration working correctly? While working on a third-party backend, I'm seeing the first iteration fail due to a limitation of the backend, then all subsequent iterations fail with:
```
Traceback (most recent call last):
  File "/home/tim/code/django/tests/expressions/tests.py", line 224, in _test_slicing_of_f_expressions
    obj = model.objects.get(name=name)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tim/code/django/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tim/code/django/django/db/models/query.py", line 649, in get
    raise self.model.DoesNotExist(
expressions.models.Company.DoesNotExist: Company matching query does not exist.
```
because the previous update didn't work.

Reply:
Good catch :dart:, check out #18398.

Reply:
Thanks for sorting this out, Mariusz!

Sorry Tim, not had much spare time of late.